src/reducers/versions.tsx

```diff
@@ -1310,17 +1322,24 @@ const reducer: Reducer<VersionsState, ActionType<typeof actions>> = (
         throw new Error(`Version missing for versionId: ${versionId}`);
       }
 
-      const { expandedPaths } = version;
+      const { expandedPaths } = state;
+
+      if (expandedPaths === undefined) {
```

Comment:
I considered this before, but finally choose to use `undefined`.

The initial `expandedPaths` needs to be a special one, so that we can know if the old `expandedPaths` should be preserved or not when load a new version.

Using empty array as initial value works in this case, since `expandedPaths` always have at least one element `~ROOT~` after initialized. However, it looks bad to have such implicitly dependence on the fact that we have `~ROOT~` which cannot moved out of `expandedPaths`. And It is not that readable to use `expandedPaths.length() === 0` to determine if it is initialized.

> Would that work for the other states?

Do you mean `selectedPath` and `visibleSelectedPath`? I guess they need to be defined as `string|undefined`.

Reply:
> The initial `expandedPaths` needs to be a special one, so that we can know if the old `expandedPaths` should be preserved or not when load a new version.

I don't see anything resetting `expandedPaths` to `undefined` after loading a version so does switching versions work as it should? After https://github.com/mozilla/addons-code-manager/pull/1204#discussion_r343889489 gets addressed then I can look at that scenario in more detail.

> > Would that work for the other states?
> 
> Do you mean `selectedPath` and `visibleSelectedPath`? I guess they need to be defined as `string|undefined`.

I was just asking if using `string[]` for `expandedPaths` would work for the other features we need to support.

Reply:
> I was just asking if using string[] for expandedPaths would work for the other features we need to support.

Yes. We only have `expandedPaths === undefined` before loading the first version (we do not reset it to `undefined`), so using [] as initial value should not break other features.